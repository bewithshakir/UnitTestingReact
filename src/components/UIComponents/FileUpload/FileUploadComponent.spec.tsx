import { shallow } from 'enzyme';
import FileUploadComponent from './FileUpload.component';

describe('Rendering of File Upload Component', () => {

    it('File Upload component Snapshot testing', () => {
        const component = shallow(<FileUploadComponent onDrop={() => jest.fn()} />);
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it('Renders children properly', () => {
        const component = shallow((<FileUploadComponent onDrop={() => jest.fn()}>
            <button>Browse</button>
        </FileUploadComponent>));
        
        expect(component).toBeDefined();
        expect(component.contains(<button>Browse</button>)).toBeTruthy();
    });

});