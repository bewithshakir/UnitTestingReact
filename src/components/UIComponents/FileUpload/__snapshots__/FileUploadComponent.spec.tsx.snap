// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering of File Upload Component File Upload component Snapshot testing 1`] = `
<Fragment>
  <div
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <input
      autoComplete="off"
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
  </div>
</Fragment>
`;
